tests/xtensor/test_shape.py

```diff
@@ -369,3 +380,195 @@ def test_squeeze_errors():
+    xr_assert_allclose(res, expected)
+
+    # Test behavior with symbolic size > 1
+    # NOTE: This test documents our current behavior where expand_dims broadcasts to the requested size.
+    # This differs from xarray's behavior where expand_dims always adds a size-1 dimension.
+    size_sym_4 = scalar("size_sym_4", dtype="int64")
+    x = xtensor("x", dims=("a", "b"), shape=(2, 3))
+    y = expand_dims(x, "batch", size=size_sym_4)
+    fn = xr_function([x, size_sym_4], y, on_unused_input="ignore")
+    x_test = xr_arange_like(x)
+    res = fn(x_test, 4)
+    # Our current behavior: broadcasts to size 4
+    expected = x_test.expand_dims({"batch": 4})
+    xr_assert_allclose(res, expected)
+    # xarray's behavior would be:
+    # expected = x_test.expand_dims("batch")  # always size 1
+    # xr_assert_allclose(res, expected)
+
+    # Test using symbolic size from a reduction operation
+    x = xtensor("x", dims=("a", "b"), shape=(2, 3))
+    reduced = x.sum("a")  # shape: (b: 3)
+    y = expand_dims(x, "batch", size=reduced.sizes["b"])
+    fn = xr_function([x], y)
+    x_test = xr_arange_like(x)
+    res = fn(x_test)
+    expected = x_test.expand_dims({"batch": 3})  # 3 is the size of dimension "b"
+    xr_assert_allclose(res, expected)
+
+    # Test chaining expand_dims with symbolic sizes
+    x = xtensor("x", dims=("a",), shape=(2,))
+    y = expand_dims(x, "b", size=x.sizes["a"])  # shape: (a: 2, b: 2)
+    z = expand_dims(y, "c", size=y.sizes["b"])  # shape: (a: 2, b: 2, c: 2)
+    fn = xr_function([x], z)
+    x_test = xr_arange_like(x)
+    res = fn(x_test)
+    expected = x_test.expand_dims({"b": 2}).expand_dims({"c": 2})
+    xr_assert_allclose(res, expected)
+
+    # Test bidirectional broadcasting with symbolic sizes
+    x = xtensor("x", dims=("a",), shape=(2,))
+    y = xtensor("y", dims=("b",), shape=(3,))
+    # Expand x with size from y, then add y
+    expanded = expand_dims(x, "b", size=y.sizes["b"])
+    z = expanded + y  # Should broadcast x to match y's size
+    fn = xr_function([x, y], z)
+    x_test = xr_arange_like(x)
+    y_test = xr_arange_like(y)
+    res = fn(x_test, y_test)
+    expected = x_test.expand_dims({"b": 3}) + y_test
+    xr_assert_allclose(res, expected)
+
+
+def test_expand_dims_errors():
+    """Test error handling in expand_dims."""
+
+    # Expanding existing dim
+    x = xtensor("x", dims=("city",), shape=(3,))
+    y = expand_dims(x, "country")
+    with pytest.raises(ValueError, match="already exists"):
+        expand_dims(y, "city")
+
+    # Size = 0 is invalid
+    with pytest.raises(ValueError, match="size must be.*positive"):
+        expand_dims(x, "batch", size=0)
+
+    # Invalid dim type
+    with pytest.raises(TypeError):
+        expand_dims(x, 123)
+
+    # Invalid size type
+    with pytest.raises(TypeError):
+        expand_dims(x, "new", size=[1])
+
+    # Duplicate dimension creation
+    y = expand_dims(x, "new")
+    with pytest.raises(ValueError):
```

Comment:
match the expected error messages to be sure you are triggering the branch you care about. Sometimes you are testing an earlier error and can't tell because of only checking for ValueError/TypeError